### PR TITLE
Remove support for THP in Redis module - recommend forge module

### LINKED ISSUE #372: After a reboot THP (Transparent Huge Pages) are enabled again
## Affected Puppet, Ruby, OS and module versions/distributions

current

## How to reproduce (e.g Puppet code you use)

```
contain '::redis::administration'

Class['::redis::administration']
-> Class['::redis']
```

## What are you seeing

On a first run THP are being disabled before Redis service is started, so it affects Redis.

However if you reboot the server then Redis will start with THP enabled as it's not disabled in a persistent way.

## What behaviour did you expect instead

THP to be disabled in a way that will survive the server restart.

### PR DESCRIPTION
**Pull request by dom-nie**
#### Pull Request (PR) description
After discussing #384 with @ekohl  I have reached the conclusion that instead of further complicating this module, it would be best to simplify it and recommend the forge module for disabling THP.
I greatly appreciate all the feedback.

#### This Pull Request (PR) fixes the following issues
Closes #372

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -886,25 +886,28 @@ Default value: `$redis::params::minimum_version`
 
 ### `redis::administration`
 
-Allows various adminstrative settings for Redis
+Allows various administrative settings for Redis
 As documented in the FAQ and https://redis.io/topics/admin
 
+For disabling Transparent Huge Pages please use forge module:
```
Comment (gcoxmoz) [inline REFERENCE.md:892]: nitpick/suggestion.  Your change eliminates `disable_thp`, and also all `thp` strings.  How about you make this line "For disabling Transparent Huge Pages (THP), please" etc etc.

No pressure, feel free to ignore.

**Review by kenyon — APPROVED** [review]

File: REFERENCE.md (hunk 1 above)
Comment (dom-nie) [reply REFERENCE.md]: I actually think you are totally right, so I have added THP to make sure it's easier to search for.

File: REFERENCE.md [hunk 2]
```diff
@@ -886,25 +886,28 @@ Default value: `$redis::params::minimum_version`
 
 ### `redis::administration`
 
-Allows various adminstrative settings for Redis
+Allows various administrative settings for Redis
 As documented in the FAQ and https://redis.io/topics/admin
 
+For disabling Transparent Huge Pages (THP), please use forge module:
+https://github.com/alexharv074/puppet-disable_transparent_hugepage
```
Comment (alexjfisher) [inline REFERENCE.md:893]: Maybe link to the forge module instead??
https://forge.puppet.com/modules/alexharvey/disable_transparent_hugepage

Comment (ekohl) [reply REFERENCE.md:893]: I personally haven't used this module. Do want to endorse it in this way or tone it down to "use separate module such as ..."?

File: REFERENCE.md [hunk 3]
```diff
@@ -886,25 +886,28 @@ Default value: `$redis::params::minimum_version`
 
 ### `redis::administration`
 
-Allows various adminstrative settings for Redis
+Allows various administrative settings for Redis
```
Comment (ekohl) [inline REFERENCE.md:889]: You should fix the typo in the file itself and then regenerate `REFERENCE.md` using `bundle exec rake strings:generate:reference`.

**Review by ekohl — COMMENTED** [review]
Overall :+1: for dropping this functionality.

File: REFERENCE.md (hunk 3 above)
Comment (dom-nie) [reply REFERENCE.md:883]: Okay, moved the comment regarding THP to the class itself. Generated reference file but is this many changes expected @ekohl ?

File: REFERENCE.md (hunk 2 above)
Comment (dom-nie) [reply REFERENCE.md]: Oh, I used both of your recommendations here.

File: REFERENCE.md [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
-
 <!-- DO NOT EDIT: This document was generated by Puppet Strings -->
 
 ## Table of Contents
 
-### Classes
+**Classes**
 
-#### Public Classes
+_Public Classes_
```
Comment (ekohl) [inline REFERENCE.md:8]: Could it be that you have an old version of strings installed and that it's using an old template?

File: manifests/administration.pp [hunk 5]
```diff
@@ -1,18 +1,18 @@
-# Allows various adminstrative settings for Redis
-# As documented in the FAQ and https://redis.io/topics/admin
+# Allows various administrative settings for Redis
```
Comment (ekohl) [inline manifests/administration.pp:1]: Perhaps this should be an explicit summary to keep it short.
```suggestion
# @summary Allows various administrative settings for Redis
```

File: REFERENCE.md (hunk 4 above)
Comment (dom-nie) [reply REFERENCE.md]: You are right, I have updated this pull request with the latest version.

**Review by ekohl — APPROVED** [review]
Looks good to me. Should we squash this in a single commit? Can be done on merge.

**Comment by dom-nie** [comment]
@ekohl  Thanks for all the review. I changed it to a single commit.

**Pull request merged**